contracts/colony/ColonyExpenditure.sol

```diff
@@ -337,6 +360,17 @@ contract ColonyExpenditure is ColonyStorage {
 
   // Internal functions
 
+  function setExpenditurePayouts(
```

Comment:
Stack depth error when I tried to cast IColonyNetwork.

Reply:
Can we adjust the modifiers on this function, then? `expenditureDraft` seems like it could subsume the `expenditureExists` functionality in to it (checking for an expenditure in draft state that doesn't exist makes no sense, so we're never going to want to do it), and then that frees up the stack slot that we need.

With solidity 8, we have the option of using the Yul IR pipeline to solve this, as well, though there are potentially issues around our assembly usage that I am yet to properly investigate.

Reply:
Relatedly, `expenditureSelfOrOwner` should really be `expenditureOwnerOrSelf`, right? :sweat_smile: